src/Components/Components/src/Forms/InputComponents/InputDate.cs

```diff
@@ -0,0 +1,108 @@
+
+namespace Microsoft.AspNetCore.Components.Forms
+{
+    /// <summary>
+    /// An input component for editing date values.
+    /// Supported types are <see cref="DateTime"/> and <see cref="DateTimeOffset"/>.
+    /// </summary>
+    public class InputDate<T> : InputBase<T>
+    {
+        const string dateFormat = "yyyy-MM-dd"; // Compatible with HTML date inputs
+
+        [Parameter] string ParsingErrorMessage { get; set; } = "The {0} field must be a date.";
+
+        /// <inheritdoc />
+        protected override void BuildRenderTree(RenderTreeBuilder builder)
+        {
+            base.BuildRenderTree(builder);
+            builder.OpenElement(0, "input");
+            builder.AddAttribute(1, "type", "date");
+            builder.AddAttribute(2, "class", FieldClass);
+            builder.AddAttribute(3, "value", BindMethods.GetValue(CurrentValueAsString));
+            builder.AddAttribute(4, "onchange", BindMethods.SetValueHandler(__value => CurrentValueAsString = __value, CurrentValueAsString));
+            builder.CloseElement();
+        }
+
+        /// <inheritdoc />
+        protected override string FormatValueAsString(T value)
+        {
+            switch (value)
+            {
+                case DateTime dateTimeValue:
+                    return dateTimeValue.ToString(dateFormat);
+                case DateTimeOffset dateTimeOffsetValue:
+                    return dateTimeOffsetValue.ToString(dateFormat);
+                default:
+                    return string.Empty; // Handles null for Nullable<DateTime>, etc.
+            }
+        }
+
+        /// <inheritdoc />
+        protected override bool TryParseValueFromString(string value, out T result, out string validationErrorMessage)
+        {
+            // Unwrap nullable types. We don't have to deal with receiving empty values for nullable
+            // types here, because the underlying InputBase already covers that.
+            var targetType = Nullable.GetUnderlyingType(typeof(T)) ?? typeof(T);
+
+            bool success;
+            if (targetType == typeof(DateTime))
+            {
+                success = TryParseDateTime(value, out result);
+            }
+            else if (targetType == typeof(DateTimeOffset))
+            {
+                success = TryParseDateTimeOffset(value, out result);
+            }
+            else
+            {
+                throw new InvalidOperationException($"The type '{targetType}' is not a supported date type.");
+            }
+
+            if (success)
+            {
+                validationErrorMessage = null;
+                return true;
+            }
+            else
+            {
+                validationErrorMessage = string.Format(ParsingErrorMessage, FieldIdentifier.FieldName);
+                return false;
+            }
+        }
+
+        static bool TryParseDateTime(string value, out T result)
+        {
+            var success = DateTime.TryParse(value, out var parsedValue);
```

Comment:
As noted in a bug that was filed recently, we're going to have to review all of these for correct usage of CultureInfo. (not asking you to solve it in this PR).